core/server/worker/src/main/java/alluxio/worker/page/PagedBlockReader.java

```diff
@@ -61,19 +61,25 @@ public class PagedBlockReader extends BlockReader {
    * @param ufsInStreamCache a cache for the in streams from ufs
    * @param conf alluxio configurations
    * @param blockId block id
+   * @param offset initial offset within the block to begin the read from
    * @param ufsBlockOptions options to open a ufs block
    */
   public PagedBlockReader(CacheManager cacheManager, UfsManager ufsManager,
                           UfsInputStreamCache ufsInStreamCache,
                           AlluxioConfiguration conf,
-                          long blockId, Protocol.OpenUfsBlockOptions ufsBlockOptions) {
-
+                          long blockId, long offset, Protocol.OpenUfsBlockOptions ufsBlockOptions) {
+    if (offset < 0 || offset > ufsBlockOptions.getBlockSize()) {
```

Comment:
How about using IllegalArgumentException?  I think IndexOutOfBoundsException is designed to array or string.

I might prefer to use Preconditions.checkArgument in this case

Reply:
how about OutOfRangeRuntimeException or InternalRuntimeException? I would personally think IndexOutofBounds is more aligned with the original java API but here if we pass in the wrong parameter then it's an internal code bug so Internal Exception makes sense as well.

Reply:
Changed to use `Preconditions.checkArgument`.